contracts/airdropsSMCampaign/Existing1155NftDropSMCampaign.sol

```diff
@@ -0,0 +1,180 @@
+import "../interfaces/AirdropInfoSMCampaign.sol";
+import "../interfaces/IAirBroFactory.sol";
+
+/// @title Airdrops existing ERC1155 tokens for airdrop recipients
+contract Existing1155NftDropSMCampaign is AirdropInfoSMCampaign, AirdropMerkleProof, IERC1155Receiver, Ownable {
+    IERC1155 public immutable rewardedNft;
+    IERC1155 public immutable rewardToken;
+    uint256 public immutable rewardTokenId;
+    uint256 public immutable tokensPerClaim;
+    uint256 public immutable totalAirdropAmount;
+
+    bool public airdropFunded = false;
+    uint256 public airdropFundBlockTimestamp;
+    address internal airdropFundingHolder;
+    address public immutable airBroFactoryAddress;
+
+    event Claimed(address indexed claimer);
+    event AirdropFunded(address contractAddress);
+    event MerkleRootChanged(bytes32 merkleRoot);
+    // event isEligibleForReward(bool);
+
+    error NotOwner();
+    error AirdropStillInProgress();
+    error AlreadyRedeemed();
+    error AlreadyFunded();
+    error InsufficientAmount();
+    error InsufficientLiquidity();
+    error AirdropExpired();
+    error NotAdmin();
+    error NotEligible();
+
+
+    mapping(address => bool) public hasClaimed;
+
+    // The root hash of the Merle Tree we previously generated in our JavaScript code. Remember
+    // to provide this as a bytes32 type and not string. Ox should be prepended.
+    bytes32 public merkleRoot;
+
+    uint256 public immutable airdropDuration;
+    uint256 public immutable airdropStartTime;
+    uint256 public immutable airdropFinishTime;
+
+    modifier onlyAdmin(){
+        if(msg.sender != IAirBroFactory(airBroFactoryAddress).admin()) revert NotAdmin();
+        _;
+    }
+
+    constructor(
+        address _rewardedNft,
+        address _reward1155Nft,
+        uint256 _tokensPerClaim,
+        uint256 _tokenId,
+        uint256 _totalAirdropAmount,
+        uint256 _airdropDuration,
+        address _airBroFactoryAddress
+    ) {
+        rewardedNft = IERC1155(_rewardedNft);
+        tokensPerClaim = _tokensPerClaim;
+        totalAirdropAmount = _totalAirdropAmount;
+        rewardToken = IERC1155(_reward1155Nft);
+        rewardTokenId = _tokenId;
+        airdropDuration = _airdropDuration * 1 days;
+        airdropStartTime = block.timestamp;
+        airdropFinishTime = block.timestamp + airdropDuration;
+        airBroFactoryAddress = _airBroFactoryAddress;
+
+    }
+
+    /// @notice Sets the merkleRoot - can only be done if admin (different from the contract owner)
+    /// @param _merkleRoot - The root hash of the Merle Tree
+    function setMerkleRoot(bytes32 _merkleRoot) external onlyAdmin {
+        merkleRoot = _merkleRoot;
+        emit MerkleRootChanged(_merkleRoot);
+    }
+
+    /// @notice Allows the airdrop creator to provide funds for airdrop reward
+    function fundAirdrop() external {
+        if (rewardToken.balanceOf(msg.sender, rewardTokenId) < totalAirdropAmount) revert InsufficientAmount();
```

Comment:
This is nice way, but not really needed.
safeTransfer will fail if the amount is less than `totalAirdropAmount` and it will revert

Reply:
@bxmmm1 @GotchaGuy This line has been removed. However, this means that if Alice decides to try to fund a contract with 1 DAI (when the airdropAmount is 10 DAI) the variables (L83-L85) would all be set to not allow anyone else to fund the airdrop again after the function is executed
https://github.com/MVPWorkshop/Airbro-contracts/blob/9c8c29165d538b7278badbfc69f65707206f3fa6/contracts/airdropsSMCampaign/Existing1155NftDropSMCampaign.sol#L81-L85
but then the safeTransfer would be reverted at line 86, and there would be not way to fund the contract from that point because of the previously set variables

https://github.com/MVPWorkshop/Airbro-contracts/blob/9c8c29165d538b7278badbfc69f65707206f3fa6/contracts/airdropsSMCampaign/Existing1155NftDropSMCampaign.sol#L86-L87

I think we might need to return the check to make sure that the user has the totalAirdropAmount in funds, and to check that they have approved the spending of the totalAirdropAmount to the airdrop SC.

Reply:
Great catch! @nlukic97  You're absolutely right. The check for enough funds has to happen before airdropFunded is set to true

Reply:
You are right, that totally slipped my mind.